### ISSUE TITLE
[Bug] MSVCP140.dll and / or VCRUNTIME140_1.dll errors

### ISSUE THREAD

**Issue by StanleyDudek** [BeamMP/BeamMP-Server#141]
**Fill out general information**
OS: Windows 7, unknown Windows-based VPS, unknown linux using wine
BeamMP-Server Version: 3.1.0

**Describe the bug**
Server owners upgrading from 3.0.2 to 3.1.0 are reporting these errors at server start.

![image](https://user-images.githubusercontent.com/49531350/197557449-1f973b67-7710-4463-9f88-752933506e34.png)

![image](https://user-images.githubusercontent.com/49531350/197557359-15da18fe-251a-48fa-8194-4c64c9434492.png)

![image](https://user-images.githubusercontent.com/49531350/197557672-dc49c07c-78a4-4eb2-a736-aa98fd2f59f5.png)

**Expected behavior**
Not that

**Additional context**
Temporary fix for these so far has been https://aka.ms/vs/17/release/vc_redist.x64.exe

**Comment by lionkor** [comment]
Hi! I think I might have fixed it, who can I contact to test a fix?

**Comment by StanleyDudek** [comment]
Uhhh, I guess any server owner that newly reports the issue in the discord or forum, and who HASN'T already installed the VC redist as a fix?

**Comment by lionkor** [comment]
Okay, sounds good, thanks

**Comment by lionkor** [comment]
Doesn't work yet, help appreciated.

**Comment by Anonymous-275** [comment]
only way to fix this is to build the server as static

**Comment by lionkor** [comment]
Yes, @Anonymous-275, I know - could you look at 

https://github.com/BeamMP/BeamMP-Server/blob/rc-v3.1.1/CMakeLists.txt#L53

and the previous versions, because NONE of them worked. That is statically building, the issue isn't that I don't know that it's the fix, the issue is that it doesn't do it

**Comment by Anonymous-275** [comment]
You need these lines
https://github.com/BeamMP/BeamMP-Launcher/blob/master/CMakeLists.txt#L5-L7

**Comment by lionkor** [comment]
> You need these lines https://github.com/BeamMP/BeamMP-Launcher/blob/master/CMakeLists.txt#L5-L7

Yes, I know, they were in the server - we just need a way to toggle them off. I went through multiple iterations: The first was to simply put the code you sent into an if(), but that didn't work (no clue why). 

Those lines are they are in the launcher are NOT how you do it with modern CMake, which we use in the Server, and so I put those lines I linked to in my previous comment. That's how you pick whether to use /MT or /MD or any of the debug variants.

I'm specifically looking to fix the solution as it is, because it is supposed to do the **SAME** thing the launcher does in the link you sent.

This is documented in CMake's documentation, and this is the correct way to do it (the way we do it currently in the server, see my link). It just seems not to work, and I'd love to know why. I cannot afford a windows license, or afford the time to download a VM, or afford really any work associated with this, so I hoped we could get one of our 4 developers who actually **ARE** on windows to look into this.

We need /MT, and toggleable, basically.

**Comment by lionkor** [comment]
https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html

![image](https://user-images.githubusercontent.com/29932116/198887460-821d7b02-9327-4258-82de-8061bcd50c04.png)

**Comment by lionkor** [comment]
![image](https://user-images.githubusercontent.com/29932116/198887476-e03b7d4e-1d4a-45a4-9c82-4f2e942fd0db.png)

**Comment by lionkor** [comment]
Maybe the vcpkg cmake is wildly out of date and doesn't support this? I wouldn't be surprised

**Comment by Anonymous-275** [comment]
you need to have all the libraries from vcpkg be static as well but you need that flag to be always active since as soon as you need or want it to not be you will have to link to that DLL there is no other way sadly.

**Comment by O1LER** [comment]
Issue persists with [v3.1.2](https://github.com/BeamMP/BeamMP-Server/releases/tag/v3.1.2)

![image](https://github.com/BeamMP/BeamMP-Server/assets/44237025/a21cc492-1caf-48b7-9cda-63398f0e9ccd)

Please keep the note and link to the Visual C++ redistributable as with [v3.1.1](https://github.com/BeamMP/BeamMP-Server/releases/tag/v3.1.1)
